api/routes/tasks.py

```diff
@@ -1,25 +1,82 @@
+import datetime
+
 from app import app
 from models import db
 from models.task import Task
+from models.slot import Slot
 from util.response import json_response
 from util.user_id import with_user_id
 from flask import request, Response
 from sqlalchemy.dialects import postgresql
 
 
-@app.route("/api/tasks", methods=["GET"])
+@app.route("/api/task/get-all", methods=["GET"])
 @with_user_id
 def get_tasks(user_id):
     query = db.select(Task).filter(Task.user_id == user_id).order_by(Task.created)
     tasks = db.session.execute(query).scalars().all()
-    return json_response([{"name": t.title, "id": t.id} for t in tasks])
+    return json_response(
+        [
+            {
+                "name": t.title,
+                "id": t.id,
+                "description": t.description,
+                "complete": t.complete,
+                "duration": t.duration,
+            }
+            for t in tasks
+        ]
+    )
+
+
+@app.route("/api/task/update", methods=["PUT"])
+@with_user_id
+def update_task(user_id):
+    # body has id and possibly new values for all task params
+    body = request.get_json()
+
+    if "name" in body:
+        db.update(Task).where(Task.user_id == user_id and Task.id == body["id"]).values(
+            name=body["name"]
+        )
 
+    if "description" in body:
+        db.update(Task).where(Task.user_id == user_id and Task.id == body["id"]).values(
+            description=body["description"]
+        )
 
-@app.route("/api/tasks", methods=["POST"])
+    if "duration" in body:
+        db.update(Task).where(Task.user_id == user_id and Task.id == body["id"]).values(
+            duration=body["duration"]
+        )
+
+    if "complete" in body:
+        # find new complete val. If we just completed, its the id of the latest work
+        # slot, else it's none/null
+        new_complete = None
+        if body["complete"]:
+            # find the latest work slot
+            now = datetime.datetime.now()
```

Comment:
I wonder if we should optionally allow to specify the current time in the request.
This would make the "debug mode" thing much easier on our part

Reply:
we can probably pass unix time in seconds or something

Reply:
I think it'd be better for us to make a wrapper around the datetime library so that any call to datetime, whether made by  use or sqlalchemy, can be controlled by a slider or something